src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -14,6 +14,48 @@
 
 logger = logging.getLogger(__name__)
 
+ServiceSpecs = TypeVar('ServiceSpecs', bound=ServiceSpec)
+
+
+class CephadmDaemonSpec(Generic[ServiceSpecs]):
```

Comment:
I would like to have a generic  <ContainerSpec>, and then a <CephadmDaemonSpec> with  a <ContainerSpec> attribute.

Things to have in the generic "ContainerSpec": ( the kind of things we need to launch a container):
I think we should have all the things that can be used in a `podman run command`... but a good start will be:
image: The image to use for the container
config_file: location + content
mounts: storage in the host
ports: ports to open
name: container name
network: network mode for the container
command: command to execute

Reply:
hm. is this something for cephadm or for mgr/cephadm? right now, this code lives in the binary.